### PR TITLE
Move Aldryn Apphooks Reload into core

### PR DESCRIPTION
**Pull request by mkoistinen**
Apphooks reload in core CMS with tests and documentation.

### CODE REVIEW

File: cms/signals/__init__.py [hunk 1]
```diff
@@ -26,10 +27,21 @@
 post_unpublish = Signal(providing_args=["instance", "language"])
 
 # fired if a public page with an apphook is added or changed
-urls_need_reloading = Signal(providing_args=[])
+cms_urls_need_reloading = Signal(providing_args=[])
```
Comment (yakky) [inline cms/signals/__init__.py]: I think we should not change the signal name without a deprecation period, and in a patch release

File: cms/signals/apphook.py [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
 from django.core.urlresolvers import clear_url_caches
 from django.core.signals import request_finished
 from cms.models import Title
+from cms.utils.apphook_reload import mark_urlconf_as_changed
 
 
 DISPATCH_UID = 'cms-restart'
 
 
+def trigger_server_restart(**kwargs):
```
Comment (yakky) [inline cms/signals/apphook.py]: This looks the same as https://github.com/divio/django-cms/pull/4348/files#diff-643105bbf8c6920853343a09e34d38acR38

File: cms/utils/apphook_reload.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals, absolute_import
```
Comment (yakky) [inline cms/utils/apphook_reload.py]: We should conform to the rest of the codebase and not use this, ATM (to be clear, I think we should update all the codebase to use this and print_function, but we should be consistent)

File: CHANGELOG.txt [hunk 4]
```diff
@@ -437,3 +437,8 @@ Please see Install/2.4 release notes *before* attempting to upgrade to version 2
 
 - Fix placeholder cache invalidation under some circumstances
 - Update translations
+
+
+=== 3.1.3 (2015-xx-xx) ===
```
Comment (yakky) [inline CHANGELOG.txt]: This should be 3.2, right?

File: cms/signals/__init__.py [hunk 5]
```diff
@@ -26,10 +27,21 @@
...
-    urls_need_reloading.connect(debug_server_restart)
+    cms_urls_need_reloading.connect(debug_server_restart)
+
+
+def trigger_server_restart(**kwargs):
+    mark_urlconf_as_changed()
+
+cms_urls_need_reloading.connect(
+    trigger_server_restart,
+    dispatch_uid='aldryn-apphook-reload-handle-urls-need-reloading'
```
Comment (yakky) [inline cms/signals/__init__.py]: I'll change this to ``cms-reload-handle-urls-need-reloading``

File: cms/signals/apphook.py (hunk 2 above)
Comment (mkoistinen) [inline cms/signals/apphook.py]: True. Sorted.

**Comment by yakky** [comment]
@mkoistinen :+1:

**Pull request merged**